### PR TITLE
Added a TS declaration file.

### PR DESCRIPTION
**Pull request by heksesang**
Here is my attempt at creating a .d.ts file to describe the types of the library. Adding such a file would make it easier to use the library in Typescript projects.

### CODE REVIEW

File: dist/jso.d.ts [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+    isRequired: Boolean
+  ): ConfigOptions[K]
+}
+
+export declare interface ConfigOptions {
+  providerID?: string
+  client_id: string
+  client_secret?: string
+  authorization: string
+  token: string
```
Comment (hubgit) [inline dist/jso.d.ts:89]: `token` doesn't seem to be a valid option for the constructor?

https://github.com/andreassolberg/jso#initializing-the-client

Comment (hubgit) [reply dist/jso.d.ts:89]: Never mind, apparently it's the endpoint URL for exchanging a code for an access token:
https://github.com/andreassolberg/jso/issues/89

Comment (hubgit) [reply dist/jso.d.ts:89]: It is at least optional, though.

**Comment by AntonioRedondo** [comment]
@andreassolberg no merging of this pull request? Having these TypeScript types for the library would be fantastic.

**Pull request closed without merging**